From 3b6ee0162ac22e569cb00a8aee5c026c86147956 Mon Sep 17 00:00:00 2001
From: Tobias Gruetzmacher <tobias-git@23.gs>
Date: Sun, 22 Mar 2015 18:43:11 +0100
Subject: [PATCH] Cast basic_istream to bool.

This shouldn't be necessary as far as I understand it (istream has an
operator bool()), but clang 3.4 with libc++ insists that it can't cast
basic_istream to bool. This may be a bug in libc++ or my bad
understanding of C++ cast rules, but changing it doesn't hurt.
---
 game/songparser.hh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/game/songparser.hh b/game/songparser.hh
index 5196c8e..21d7e16 100644
--- a/game/songparser.hh
+++ b/game/songparser.hh
@@ -33,7 +33,7 @@ class SongParser {
 	Song& m_song;
 	std::stringstream m_ss;
 	unsigned int m_linenum;
-	bool getline(std::string& line) { ++m_linenum; return std::getline(m_ss, line);}
+	bool getline(std::string& line) { ++m_linenum; return (bool) std::getline(m_ss, line);}
 	bool m_relative;
 	double m_gap;
 	double m_bpm;
